### PR TITLE
Support KeyManagerFactory if compiled against OpenSSL < 1.0.2 but usi…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ng OpenSSL >= 1.0.2 at runtime.

Motivation:

On CentOS and Debian we compile against old versions of OpenSSL to archive maximal compatibility. Because of this we should ensure we can make use of features needed to support KeyManagerFactory when a newer version of OpenSSL is used during runtime.

Modifications:

Check on runtime the used OpenSSL version and based on this allow to use the function calls or not.

Result:

Fixes https://github.com/netty/netty-tcnative/issues/273.

### CODE REVIEW

File: openssl-dynamic/src/main/c/ssl.c [hunk 1]
```diff
@@ -664,7 +664,7 @@ TCN_IMPLEMENT_CALL(jint, SSL, initialize)(TCN_STDARGS, jstring engine)
     }
 
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
-    if (SSLeay() < 0x0090700L) {
+    if (OpenSSL_version_num() < 0x0090700L) {
```
Comment (normanmaurer) [inline openssl-dynamic/src/main/c/ssl.c]: FYI: OpenSSL_version_num is the same as `SSLeay` but the "new API". If an old version of OpenSSL is used we delegate internally.

**Comment by normanmaurer** [comment]
Verified locally by build on debian 7 and run on recent version of ubuntu. Works now as expected 🎉

File: openssl-dynamic/src/main/c/ssl.c [hunk 2]
```diff
@@ -1847,10 +1854,16 @@ TCN_IMPLEMENT_CALL(void, SSL, setKeyMaterialServerSide)(TCN_STDARGS, jlong ssl,
 
 TCN_IMPLEMENT_CALL(void, SSL, setKeyMaterialClientSide)(TCN_STDARGS, jlong ssl, jlong certOut, jlong keyOut, jlong chain, jlong key)
 {
-#if OPENSSL_VERSION_NUMBER < 0x10002000L || defined(LIBRESSL_VERSION_NUMBER)
-    // Only supported on openssl 1.0.2+
-    tcn_Throw(e, "Not supported");
+#if defined(LIBRESSL_VERSION_NUMBER)
+    tcn_Throw(e, "Not supported with LibreSSL");
 #else
+#ifndef OPENSSL_IS_BORINGSSL
```
Comment (carl-mastrangelo) [inline openssl-dynamic/src/main/c/ssl.c]: this is somewhat confusing.  You might consider putting in a comment which `endif` is associated.  e.g.

```
#ifndef OPENSSL_IS_BORINGSSL
...
#endif // OPENSSL_IS_BORINGSSL
```

Comment (normanmaurer) [reply openssl-dynamic/src/main/c/ssl.c]: done

**Pull request merged**